Frameworks/Foundation/NSInvocationOperation.mm

```diff
@@ -14,30 +14,131 @@
 //
 //******************************************************************************
 
-#import <StubReturn.h>
-#import <Starboard.h>
 #import <Foundation/NSInvocationOperation.h>
+#import <Starboard/SmartTypes.h>
+#import <NSLogging.h>
+#import <memory>
 
 NSString* const NSInvocationOperationVoidResultException = @"NSInvocationOperationVoidResultException";
 NSString* const NSInvocationOperationCancelledException = @"NSInvocationOperationCancelledException";
 
+@interface NSInvocationOperation () {
+    StrongId<NSException> _caughtException;
+    StrongId<id> _result;
+}
+@end
+
 @implementation NSInvocationOperation
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithTarget:(id)target selector:(SEL)sel object:(id)arg {
-    UNIMPLEMENTED();
-    return StubReturn();
+    NSMethodSignature* methodSignature = [target methodSignatureForSelector:sel];
+    if (!methodSignature) {
+        [self release];
+        return nil;
+    }
+
+    if (methodSignature.numberOfArguments > 3) {
```

Comment:
fwiw it's because target and selector are the first two arguments in a NSMethodSignature